src/types.ts

```diff
@@ -41,7 +41,8 @@ export type InterfaceProperty =
   | PrimitiveType
   | [PrimitiveType]
   | UnionType<PrimitiveType>
-  | InterfaceDefinition;
+  | InterfaceDefinition
+  | boolean;
```

Comment:
You forgot to add `string` (which is returned by `convertToPrimitiveRepresentation`). But since `ts-morph` library may add better types in the future, I would recommend importing the type:

```typescript
import { Type } from 'ts-morph';
```

and then add this to declaration:

```typescript
export type InterfaceProperty =
  | PrimitiveType
  | [PrimitiveType]
  | UnionType<PrimitiveType>
  | InterfaceDefinition
  | boolean
  | ReturnType<Type['getText']>
```
```

Reply:
I added `ReturnType<Type['getText']>` to InterfaceProperty type